tests/devices/unit_tests/i24/test_pmac.py

```diff
@@ -106,10 +105,36 @@ async def update_counter():
         lambda *args, **kwargs: asyncio.create_task(update_counter()),  # type: ignore
     )
 
+    set_mock_value(fake_pmac.counter_time, 0.1)
     await fake_pmac.run_program.kickoff()
-    await fake_pmac.run_program.complete(counter_timeout=0.1)
+    await fake_pmac.run_program.complete()
+
+    assert await fake_pmac.counter.get_value() == 3
+
+
+async def test_counter_refresh_timeout(fake_pmac: PMAC, RE):
+    async def update_counter():
+        set_mock_value(fake_pmac.scanstatus, 1)
+        set_mock_value(fake_pmac.counter, 0)
+        await asyncio.sleep(0.05)
+        set_mock_value(fake_pmac.counter, 1)
+        await asyncio.sleep(0.05)
+        set_mock_value(fake_pmac.counter, 2)
+        await asyncio.sleep(0.2)
+        set_mock_value(fake_pmac.counter, 3)
+        set_mock_value(fake_pmac.scanstatus, 0)
```

Comment:
Could: This can just be in a separate function to avoid the code duplication e.g.
```python
    async def update_counter(sleep_time: float):
        set_mock_value(fake_pmac.scanstatus, 1)
        set_mock_value(fake_pmac.counter, 0)
        await asyncio.sleep(0.05)
        set_mock_value(fake_pmac.counter, 1)
        await asyncio.sleep(0.05)
        set_mock_value(fake_pmac.counter, 2)
        await asyncio.sleep(sleep_time)
        set_mock_value(fake_pmac.counter, 3)
        set_mock_value(fake_pmac.scanstatus, 0)
```
Then in each test you can use:
```python
from functools import partial
    callback_on_mock_put(
        fake_pmac.pmac_string,
        lambda *args, **kwargs: asyncio.create_task(partial(update_counter, 0.2)),  # type: ignore
    )